runtime/src/main/java/io/quarkiverse/freemarker/runtime/FreemarkerBuildConfig.java

```diff
@@ -15,16 +19,120 @@
      * Comma-separated list of absolute resource paths to scan recursively for templates.
      * All tree folder from 'resource-paths' will be added as a resource.
      * Unprefixed locations or locations starting with classpath will be processed in the same way.
+     *
+     * @deprecated Use template set group of options instead ({@code quarkus.freemarker.<template-set-name>.base-path}
+     *             {@code quarkus.freemarker.<template-set-name>.includes} and
+     *             {@code quarkus.freemarker.<template-set-name>.excludes})
      */
-    @ConfigItem(defaultValue = "freemarker/templates")
-    public List<String> resourcePaths;
+    @ConfigItem
+    public Optional<List<String>> resourcePaths;
 
     /**
      * List of directives to register with format name=classname
-     * 
+     *
      * @see freemarker.template.Configuration#setSharedVariable(String, TemplateModel)
      */
     @ConfigItem
     public Map<String, String> directive;
 
+    /**
+     * The default template set
+     */
+    @ConfigItem(name = ConfigItem.PARENT)
+    public TemplateSet defaultTemplateSet;
+
+    /**
+     * Additional named template sets.
+     */
+    @ConfigDocSection
+    @ConfigDocMapKey("template-set-name")
+    @ConfigItem(name = ConfigItem.PARENT)
+    public Map<String, TemplateSet> namedTemplateSets;
+
+    @ConfigGroup
+    public static class TemplateSet {
+        /**
+         * The base path of this template set.
+         * <p>
+         * Use slash ({@code /}) as a path separator on all platforms. The value must not start with a slash.
+         * <p>
+         * {@link #includes} and {@link #excludes} are relative to {@link #basePath}.
+         * <p>
+         * Defaults:
+         * <table>
+         * <tr>
+         * <th>Option</th>
+         * <th>Default value</th>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.freemarker.base-path}</td>
+         * <td>{@code freemarker/templates}</td>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.freemarker.includes}</td>
+         * <td>{@code **}</td>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.freemarker.excludes}</td>
+         * <td>not set</td>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.<template-set-name>.freemarker.base-path}</td>
+         * <td>Empty string (root path)</td>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.<template-set-name>.freemarker.includes}</td>
+         * <td>not set</td>
+         * </tr>
+         * <tr>
+         * <td>{@code quarkus.<template-set-name>.freemarker.excludes}</td>
+         * <td>not set</td>
+         * </tr>
+         * </table>
+         * {@code quarkus.freemarker.<template-set-name>.base-path}
+         *
+         * @since 0.2.0
```

Comment:
The version of Quarkiverse Freemarker since which the option is available. I assume the next version should increment minor because this change (1) is backwards compatible and (2) adds a new config option.

Reply:
ok. it was not clear you had anticipated the next version.